apps/arena/native/physics/src/lib.rs

```diff
@@ -135,9 +135,25 @@ fn get_direction_from_positions(position_a: Position, position_b: Position) -> D
 }
 
 #[rustler::nif()]
-fn calculate_duration(position_a: Position, position_b: Position, speed: f32) -> u64 {
+fn calculate_duration(
+    position_a: Position,
+    position_b: Position,
+    speed: f32,
+    max_range: f32,
+) -> u64 {
     let len = distance_between_positions(position_a, position_b);
-    (len / speed) as u64
+    if len > max_range {
+        let mut direction = Position::sub(&position_b, &position_a);
+        direction.normalize();
+
+        let new_target = Position::add(&position_a, &Position::mult(&direction, max_range));
+
+        let len = distance_between_positions(position_a, new_target);
+
+        (len / speed) as u64
+    } else {
+        (len / speed) as u64
+    }
```

Comment:
This is what I mentioned in Skill.ex file.

Reply:
Answered above

Reply:
I'm not sure if this is the best place, we could handle it in Elixir before but we can address this later.